dll/win32/shell32/shlexec.cpp

```diff
@@ -1256,29 +1256,51 @@ HINSTANCE WINAPI FindExecutableA(LPCSTR lpFile, LPCSTR lpDirectory, LPSTR lpResu
  */
 HINSTANCE WINAPI FindExecutableW(LPCWSTR lpFile, LPCWSTR lpDirectory, LPWSTR lpResult)
 {
-    UINT_PTR retval = SE_ERR_NOASSOC;
-    WCHAR old_dir[1024];
-    WCHAR res[MAX_PATH];
+    UINT_PTR retval = 42;
```

Comment:
Why that had been changed to a magic number, as you still return `SE_ERR_NOASSOC` in line 1294?

Reply:
This value is special and not defined in PSDK headers.

Reply:
What also @SigmaTel71 wants to say is that there currently doesn't seem to be a need to initialize retval to anything, since all the code paths below are (re)initializing its value.

Reply:
> What also @SigmaTel71 wants to say is that there currently doesn't seem to be a need to initialize retval to anything
I would still initialize the variable at least to zero, but given the code context I would like to keep it initializing to `SE_ERR_NOASSOC`, eliminating one redundant return path in line 1294, making it a value that returns when no other conditions were met.

Reply:
> But then, a question for @katahiromz is this "42" value documented somewhere, as a "known" returned value by the function?

Nope. I found `"42"` in decoder:
![image](https://github.com/reactos/reactos/assets/2107452/63c58ca1-dcba-4ace-8e73-b94431a34321)

Reply:
`retval` assignment is moved.

Reply:
`SE_ERR_NOASSOC` is not `42`.